securedrop_salt/sd-proxy.sls

```diff
@@ -23,13 +24,18 @@ sd-proxy-dvm:
       - netvm: sd-whonix
       - template_for_dispvms: True
       - default_dispvm: ""
+    - features:
+    {% if d.environment == "prod" %}
+      - set:
+        - internal: 1
+    {% endif %}
     - tags:
       - add:
         - sd-workstation
         - sd-{{ sdvars.distribution }}
     - require:
-      - qvm: sd-whonix
-      - qvm: sd-small-{{ sdvars.distribution }}-template
+      - qvm: securedrop_salt.sd-whonix
+      - qvm: securedrop_salt.sd-small-{{ sdvars.distribution }}-template
```

Comment:
Based on—

https://github.com/freedomofpress/securedrop-workstation/blob/c13aeaa500dc3cbce58c45e20dbc1c5e9d07b55e/securedrop_salt/sd-proxy.sls#L13-L15

—I believe this needs to be:

```suggestion
      - sls: securedrop_salt.sd-whonix
      - sls: securedrop_salt.sd-small-{{ sdvars.distribution }}-template
```

<details><summary>Failure on previous commit</summary></details>

```
INFO:[2024-06-04-18:56:49:482843]           ID: sd-proxy-dvm
INFO:[2024-06-04-18:56:49:482867]     Function: qvm.vm
INFO:[2024-06-04-18:56:49:482894]       Result: False
INFO:[2024-06-04-18:56:49:482919]      Comment: An exception occurred in this state: Traceback (most recent call last):
INFO:[2024-06-04-18:56:49:482947]                 File "/usr/lib/python3.11/site-packages/salt/state.py", line 2423, in call
INFO:[2024-06-04-18:56:49:482971]                   ret = self.states[cdata["full"]](
INFO:[2024-06-04-18:56:49:483001]                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483025]                 File "/usr/lib/python3.11/site-packages/salt/loader/lazy.py", line 159, in __call__
INFO:[2024-06-04-18:56:49:483051]                   ret = self.loader.run(run_func, *args, **kwargs)
INFO:[2024-06-04-18:56:49:483078]                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483102]                 File "/usr/lib/python3.11/site-packages/salt/loader/lazy.py", line 1245, in run
INFO:[2024-06-04-18:56:49:483127]                   return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
INFO:[2024-06-04-18:56:49:483154]                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483179]                 File "/usr/lib/python3.11/site-packages/salt/loader/lazy.py", line 1260, in _run_as
INFO:[2024-06-04-18:56:49:483206]                   return _func_or_method(*args, **kwargs)
INFO:[2024-06-04-18:56:49:483240]                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483269]                 File "/usr/lib/python3.11/site-packages/salt/loader/lazy.py", line 1293, in wrapper
INFO:[2024-06-04-18:56:49:483297]                   return f(*args, **kwargs)
INFO:[2024-06-04-18:56:49:483321]                          ^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483346]                 File "/var/cache/salt/minion/extmods/states/ext_state_qvm.py", line 431, in vm
INFO:[2024-06-04-18:56:49:483372]                   _varargs, keywords = parse_options(kwargs[action])
INFO:[2024-06-04-18:56:49:483397]                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INFO:[2024-06-04-18:56:49:483421]                 File "/var/cache/salt/minion/extmods/states/ext_state_qvm.py", line 419, in parse_options
INFO:[2024-06-04-18:56:49:483449]                   for option in options:
INFO:[2024-06-04-18:56:49:483473]               TypeError: 'NoneType' object is not iterable
INFO:[2024-06-04-18:56:49:483501]      Started: 18:56:26.557304
INFO:[2024-06-04-18:56:49:483529]     Duration: 2409.68 ms
INFO:[2024-06-04-18:56:49:483555]      Changes:   
```

</details>

<details><summary>Failure on this commit</summary>

```
INFO:[2024-06-04-21:23:59:145074]           ID: sd-proxy-dvm
INFO:[2024-06-04-21:23:59:145099]     Function: qvm.vm
INFO:[2024-06-04-21:23:59:145126]       Result: False
INFO:[2024-06-04-21:23:59:145154]      Comment: The following requisites were not found:
INFO:[2024-06-04-21:23:59:145180]                                  require:
INFO:[2024-06-04-21:23:59:145206]                                      qvm: securedrop_salt.sd-whonix
INFO:[2024-06-04-21:23:59:145234]                                      qvm: securedrop_salt.sd-small-bookworm-template
INFO:[2024-06-04-21:23:59:145386]      Started: 21:23:37.906327
INFO:[2024-06-04-21:23:59:145417]     Duration: 0.003 ms
INFO:[2024-06-04-21:23:59:145443]      Changes:
```

</details>

Reply:
ugh. thank you @cfm !

Reply:
(update: actually see [below](https://github.com/freedomofpress/securedrop-workstation/pull/857#issuecomment-2149657488): in this case we do want to rely on just the vm creation as a prereq (qvm), not the entire sls file )